pkg/operatorsource/configuring_phase_reconciler.go

```diff
@@ -0,0 +1,120 @@
+package operatorsource
```

Comment:
I like the idea of using a sub package. If I use `reconciler` as the sub package name, then it will introduce stuttering - `reconciler.PhaseReconciler`
Instead, I propose that we name the sub package `phase`, so the usage within `operatorsource` package will look like this - `phase.Reconciler`. And then the file name within the subpackage can be `configuring.go` and such. 

In addition, if we want to use a subpackage, I will need to elevate `Datastore` and `Kubeclient` into its own packages to avoid cyclic dependencies.

Reply:
I am fine with this approach.

Reply:
...except I don't understand why you would need to elevate `Datastore`. Can't that remain where it is?

Reply:
Same reason, cyclic dependency - `operatorsource` imports `phase` and `phase` imports `operatorsource` for `Datastore`.

Reply:
Won't it be better to have a `common` package for stuff like this instead of elevating and creating new packages for every element?

Reply:
We also have to look at it from a usage perspective or how the code reads - `datastore.Reader` and `datastore.Writer` seems more concise and readable than `common.DatastoreReader`. 
In my opinion, `datastore` warrants its own package.

Reply:
You can also import the package as `c` and then call it `c.datastore` to make it more concise.